src/EnergyPlus/WeatherManager.cc

```diff
@@ -3657,16 +3683,44 @@ namespace WeatherManager {
                     TomorrowWindSpeed(TS, Hour) = LastHrWindSpeed * WtPrevHour + Wthr.WindSpeed(Hour) * WtNow;
                     TomorrowWindDir(TS, Hour) = interpolateWindDirection(LastHrWindDir, Wthr.WindDir(Hour), WtNow);
                     TomorrowHorizIRSky(TS, Hour) = LastHrHorizIRSky * WtPrevHour + Wthr.HorizIRSky(Hour) * WtNow;
-                    if (Environment(Envrn).SkyTempModel > 3 || Environment(Envrn).SkyTempModel == 0) {
-                        TomorrowSkyTemp(TS, Hour) = LastHrSkyTemp * WtPrevHour + Wthr.SkyTemp(Hour) * WtNow;
-                    }
                     TomorrowDifSolarRad(TS, Hour) =
                         LastHrDifSolarRad * WgtPrevHour + Wthr.DifSolarRad(Hour) * WgtHourNow + NextHrDifSolarRad * WgtNextHour;
                     TomorrowBeamSolarRad(TS, Hour) =
                         LastHrBeamSolarRad * WgtPrevHour + Wthr.BeamSolarRad(Hour) * WgtHourNow + NextHrBeamSolarRad * WgtNextHour;
 
                     TomorrowLiquidPrecip(TS, Hour) = LastHrLiquidPrecip * WtPrevHour + Wthr.LiquidPrecip(Hour) * WtNow;
                     TomorrowLiquidPrecip(TS, Hour) /= double(NumOfTimeStepInHour);
+                    TomorrowTotalSkyCover(TS, Hour) = LastHrTotalSkyCover * WtPrevHour + Wthr.TotalSkyCover(Hour) * WtNow;
+                    TomorrowOpaqueSkyCover(TS, Hour) = LastHrOpaqueSkyCover * WtPrevHour + Wthr.OpaqueSkyCover(Hour) * WtNow;
+
+                    Real64 InterpOpaqueSkyCover = TomorrowOpaqueSkyCover(TS, Hour); 
+                    Real64 InterpDryBulb = TomorrowOutDryBulbTemp(TS, Hour); 
+                    Real64 InterpDewPoint = TomorrowOutDewPointTemp(TS, Hour); 
+                    Real64 InterpRelHumDeci = TomorrowOutRelHum(TS, Hour) * 0.01; 
+                    Real64 InterpIRHoriz = TomorrowHorizIRSky(TS, Hour); 
+
+                    // Sky emissivity now takes interpolated timestep inputs rather than interpolated calcation esky results
+                    ESky = CalcSkyEmissivity(Environment(Envrn).SkyTempModel, InterpOpaqueSkyCover, InterpDryBulb, InterpDewPoint, InterpRelHumDeci);
+                    if (!Environment(Envrn).UseWeatherFileHorizontalIR || InterpIRHoriz >= 9999.0) {
+                        TomorrowHorizIRSky(TS, Hour) = ESky * Sigma * pow_4(InterpDryBulb + TKelvin);
+                    } else {
+                        TomorrowHorizIRSky(TS, Hour) = InterpIRHoriz;
+                    }
```

Comment:
The` Esky` location is the original code structure and logic. But after an examining context, it is possible to have it get into the if block. Only it is a little weird that is another similar if block a few lined down and they are in sync. If the if criterion of these two "ifs" are synchronously changed together in later revision, there seems to be no problem.

Reply:
The `InterpIRHoriz` is a little bit trickier. There are a few issue here. It is used both in the "if" criterion to determine if the weather file contains a valid value (9999); but it also play a role in the body of if-else.
In a sense it might be a rare occurrence that in a weather file it has valid value in one hour then it does not the other. But in that case, the original method would be messed up as well, as it still will interpolate (actually interpolated before here) `IRHoriz` as output.
Ideally if everything here is timestep-wise would be the best. We can also use the hourly `IRHoriz` for the "if" criterion but use the interpolated `IRHoriz` as output (seems this will aligned with the original method's path) for the occurrence of `IRHoriz` in the weather file is invalid (>=9999).